src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -74,6 +77,21 @@ class LogInWithShortLivedTokenPage extends Component {
             Log.info('[LoginWithShortLivedTokenPage] exitTo is workspace/new - handling new workspace creation in AuthScreens');
             return;
         }
+        this.navigateToExitRoute();
+    }
+
+    componentDidUpdate() {
+        this.navigateToExitRoute();
+    }
```

Comment:
Other things could later update this component... should we "navigate to the exit route" if they do?

Reply:
The only things that can make this component update are the props correct? So as long as we don't add more props / Onyx keys it will be fine.

What if we moved all of this into an action called "signInTransitioningUser" that would handle signing in, out, and navigating? That way instead of using componentDidUpdate to wait for changes we could use promises / Onyx callbacks to call these actions in sequence. For example,  sign out old user > wait for onyx to clear > sign in new user > navigate. I'm not sure if that's any different than what we are doing here, but please let me know what you think.

Reply:
>The only things that can make this component update are the props correct

`componentDidUpdate()` runs after a component's `render()` method is called. a change in props or state will cause the component to render, but so can other things like a parent component re-rendering (with no change at all to props or state).

>So as long as we don't add more props / Onyx keys it will be fine.

That's a good way to rephrase my concern 😄  

How do we make everyone understand that the possibility exists to add more props and for things to not be fine?

>What if we moved all of this into an action called "signInTransitioningUser" that would handle signing in, out, and navigating? That way instead of using componentDidUpdate to wait for changes we could use promises / Onyx callbacks to call these actions in sequence. For example, sign out old user > wait for onyx to clear > sign in new user > navigate. I'm not sure if that's any different than what we are doing here, but please let me know what you think.

This sounds interesting and I think it is pretty different compared to what we are doing here. It also sounds much easier to follow because the code paths will be separated cleanly and we won't have to worry about which "case" we are handling as we bounce between different lifecycle methods. Let's chat more 1:1 about this.